src/box/lua/net_box.c

```diff
@@ -1322,19 +1352,32 @@ netbox_decode_nil(struct lua_State *L, const char **data,
 
 /**
  * This helper skips a MessagePack map header and IPROTO_DATA key so that
- * *data points to the actual response content.
+ * *data points to the actual response content. It is expected that IPROTO_DATA
+ * goes first.
+ * Returns size of map.
  */
-static void
+static uint32_t
 netbox_skip_to_data(const char **data)
 {
 	assert(mp_typeof(**data) == MP_MAP);
 	uint32_t map_size = mp_decode_map(data);
-	/* Until 2.0 body has no keys except DATA. */
-	assert(map_size == 1);
-	(void)map_size;
 	uint32_t key = mp_decode_uint(data);
 	assert(key == IPROTO_DATA);
 	(void)key;
+	return map_size;
+}
+
+/**
+ * This helper skips a MessagePack map header and IPROTO_DATA key so that
+ * *data points to the actual response content. It is expected that IPROTO_DATA
+ * is the only key of the map.
+ */
+static void
+netbox_skip_to_the_only_data(const char **data)
```

Comment:
The name is terrible. Better leave `netbox_skip_to_data` as is and inline the code that decodes the header into `netbox_decode_select`. No need in a separate helper function for this.

Reply:
Done